kong.conf.default

```diff
@@ -901,6 +901,31 @@
                                  # database servers and result in less jitter
                                  # in proxy latency, with downside of longer
                                  # converge time for router updates.
+                                 
+#upstream_consistency = strict   # Defines whether this node should reload its
+                                 # upstream synchronously or asynchronously (the
+                                 # upstream is reload every time a upstream
+                                 # is updated via the Admin API or
+                                 # loading a declarative configuration file).
+                                 #
+                                 # Accepted values are:
+                                 #
+                                 # - `strict`: the upstream will be reload
+                                 #   synchronously, causing incoming requests to
+                                 #   be delayed until the reload is finished.
+                                 # - `eventual`: the upstream will be reload
+                                 #   asynchronously via a recurring background
+                                 #   job running every second inside of each
+                                 #   worker.
+
+#upstream_update_frequency = 1   # Defines how often the upstream changes are
+                                 # checked with a background job. When a change
+                                 # is detected, a new upstream will be reload. By
+                                 # default we check for changes every second.
+                                 # Raising this value will decrease the load on
+                                 # database servers and result in less jitter
+                                 # in proxy latency, with downside of longer
+                                 # converge time for upstream updates.
```

Comment:
About those new options: I hardly see a case for wanting a different consistency for the router than for the load balancer (and vice-versa). If strict consistency of the node is desired, both the router _and_ the balancer have to be set to strict. If asynchronous updates are desired, then both should use an eventual consistency.

My suggestions is to merge both of these properties into one, and deprecate the old router consistency settings. As for the name of the new properties, suggestions are welcome. I can see one of:

1. `consistency`
2. `node_consistency`
3. `proxy_state_consistency`

With 3 being my favorite so far. Again, suggestions welcome!

As for the description, we should mention that this property defines how the node reloads the "router and load balancer states" (note that imho we should use "load balancer" because "upstream" is either too vague or too close to the Kong "Upstream" entity).

The old `router_*` properties can be deprecated and fallback to setting the new `proxy_state_consistency` when specified (`conf_loader.lua` has easy support for such deprecation, see the `upstream_keepalive` property).